cpp/src/copying/gather.cu

```diff
@@ -74,17 +39,37 @@ std::unique_ptr<table> gather(table_view const& source_table,
 {
   CUDF_EXPECTS(gather_map.has_nulls() == false, "gather_map contains nulls");
 
-  std::unique_ptr<table> destination_table = cudf::type_dispatcher(gather_map.type(),
-                                                                   dispatch_map_type{},
-                                                                   source_table,
-                                                                   gather_map,
-                                                                   gather_map.size(),
-                                                                   bounds,
-                                                                   neg_indices,
-                                                                   mr,
-                                                                   stream);
+  // create index type normalizing iterator for the gather_map
+  auto map_begin = indexalator_factory::make_input_iterator(gather_map);
+  auto map_end   = map_begin + gather_map.size();
 
-  return destination_table;
+  if (bounds == out_of_bounds_policy::FAIL) {
+    cudf::size_type begin =
+      neg_indices == negative_index_policy::ALLOWED ? -source_table.num_rows() : 0;
+    cudf::size_type end = source_table.num_rows();
+    CUDF_EXPECTS(gather_map.size() == thrust::count_if(rmm::exec_policy()->on(0),
```

Comment:
Yikes!
```suggestion
    CUDF_EXPECTS(gather_map.size() == thrust::count_if(rmm::exec_policy(stream)->on(stream),
```

Reply:
Good catch. This has been in there forever. Also, this file was also missing a copyright header.